lib/capistrano/tasks/unicorn.rake

```diff
@@ -36,9 +36,15 @@ namespace :unicorn do
   desc 'Setup Unicorn initializer'
   task :setup_initializer do
     on roles :app do
-      sudo_upload! template('unicorn_init.erb'), unicorn_initd_file
-      execute :chmod, '+x', unicorn_initd_file
-      sudo 'update-rc.d', '-f', fetch(:unicorn_service), 'defaults'
+      case fetch(:init_system)
+      when :sysv
+        sudo_upload! template('unicorn_init.erb'), unicorn_initd_file
+        execute :chmod, '+x', unicorn_initd_file
+        sudo 'update-rc.d', '-f', fetch(:unicorn_service), 'defaults'
+      when :systemd
+        sudo_upload! template('unicorn_service.erb'), fetch(:unicorn_service)
```

Comment:
You are correct I have updated the path(I had already done so locally to make it work).